core/src/banking_stage.rs

```diff
@@ -363,10 +364,12 @@ impl BankingStage {
         transaction_status_sender: Option<TransactionStatusSender>,
     ) {
         let socket = UdpSocket::bind("0.0.0.0:0").unwrap();
-        let mut buffered_packets = vec![];
+        let mut buffered_packets: Vec<PacketsAndOffsets> = vec![];
         loop {
-            if !buffered_packets.is_empty() {
-                Self::process_buffered_packets(
+            let buffered_process_start = Instant::now();
+            while !buffered_packets.is_empty() {
+                let buffered_len: usize = buffered_packets.iter().map(|p| p.1.len()).sum::<usize>();
+                let decision = Self::process_buffered_packets(
```

Comment:
Quick clarification, so because the unit of processing in `process_received_packets`: https://github.com/solana-labs/solana/blob/master/core/src/banking_stage.rs#L757 is a single `Packets`. So if all the packets inside `Packets` cause `AccountInUse` on the same account, then at most one of the transactions can be processed at a time (lock seems to imply that here: https://github.com/solana-labs/solana/blob/master/runtime/src/accounts.rs#L557), but the entire `Packets` object is being deserialized/processed/rebuffered every time? 

But if the transactions causing the `AccountInUse` conflicts are spread across different `Packets`, then a single iteration will successfully process all the `Packets` without any `AccountInUse` conflicts?

Reply:
Yes, we would deserialize again, that's another optimization we could make.

>But if the transactions causing the AccountInUse conflicts are spread across different Packets, then a single iteration >will successfully process all the Packets without any AccountInUse conflicts?

I might be wrong, I don't think Packets matters, multiple Packet can be picked from multiple Packets arrays and combined into any number of entries. transactions_from_packets will just linearize the available packets and if they are chunked with duplicate accounts, then we would get those errors.

Reply:
This makes it seem like we are deserializing/processing/locking based on one `Packets` at a time: https://github.com/solana-labs/solana/blob/master/core/src/banking_stage.rs#L196-L202

Reply:
Ah, yes, that's right. We should probably combine up to a limit.